### PR TITLE
index.rst: Add hint to use --json and --format

### PR DESCRIPTION
**Pull request by andreidinu**
Added small description for --json and --format
when used with coala command. 

Closes https://github.com/coala/documentation/issues/278

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -115,6 +115,69 @@ Use the ``--format`` option if you want to use a custom format string. Both of
...
+
+|
+
+``--format`` returns a linear, one line output.
+
+Example:
+
+.. code-block:: bash
+
+    id:56116936619154071241567282368738969837:origin:PycodestyleBear
```
Comment (Nosferatul) [inline index.rst]: Maybe we should explain this example?

Comment (Nosferatul) [inline index.rst]: This id shouldn't be the same as the `id` given in the json?

File: index.rst [hunk 2]
```diff
@@ -115,6 +115,69 @@ Use the ``--format`` option if you want to use a custom format string. Both of
...
+~~~~
+
+``--json`` makes the coala output easy to read and parse. It uses
+alphabetically ordered objects, arrays, strings, values and numbers. You can
+read more about this format `here <http://www.json.org>`__.
+
+Example:
+
+.. code-block:: bash
+
```
Comment (Mixih) [inline index.rst]: Hmm, maybe an example is not the way to go as this is going to make things long and hard to read

**Comment by andreidinu** [comment]
Should I delete the first example (or both of them) and just try to write larger descriptions?

File: Users/Install.rst [hunk 3]
```diff
@@ -199,7 +199,7 @@ application also asks for the packages needed in case it does not
 have it.
 
 The requirements files ( Gemfile, requirements.txt, etc.) are in
-the coala-bears repository and you should not get them from 
+the coala-bears repository and you should not get them from
```
Comment (Adrianzatreanu) [inline Users/Install.rst]: this looks unrelated to the change from this commit

File: Users/Tutorial.rst [hunk 4]
```diff
@@ -331,7 +331,7 @@ work for all bears.
 ::
 
     # noqa
-    long_line = "This is a long line ... "
```
Comment (Adrianzatreanu) [inline Users/Tutorial.rst]: same here, totally unrelated, probably your editor edited this by mistake and you added this file by mistake

File: Users/Install.rst (hunk 3 above)
Comment (jayvdb) [inline Users/Install.rst]: how did this even happen; we have SpaceConsistencyBear enabled in the .coafile .

Comment (andreidinu) [inline Users/Install.rst]: When I run coala on my documentation directory it put it automatically. I will repair it on my next PR.

Comment (sils) [inline Users/Install.rst]: gitmate has a bug which made this possible, the coala 0.9.1 release is needed to fix that (and we should run coala on this repo since gitmate is incremental!)

File: index.rst [hunk 5]
```diff
@@ -115,6 +115,22 @@ Use the ``--format`` option if you want to use a custom format string. Both of
...
+``--json`` makes the coala output easy to read and parse. It consists of
+a collection of name/value pairs usually represented by objects and an ordered
+list of values stored in arrays. You can read more about this format `here
+<http://www.json.org>`__.
+
+
+
+``--format`` returns a linear, one line output. The separator for its elements
+will be ``:``. This option will not show all tested areas but those with
```
Comment (sils) [inline index.rst]: it won't necessarily be separated by colon, you can give a custom format string, this is the easiest to parse as you can customize the output heavily

File: index.rst [hunk 6]
```diff
@@ -115,6 +115,22 @@ Use the ``--format`` option if you want to use a custom format string. Both of
...
+Hint
+~~~~
+
+``--json`` makes the coala output easy to read and parse. It consists of
+a collection of name/value pairs usually represented by objects and an ordered
+list of values stored in arrays. You can read more about this format `here
+<http://www.json.org>`__.
+
+
+
```
Comment (sils) [inline index.rst]: same below of course

File: Users/Install.rst (hunk 3 above)
Comment (Adrianzatreanu) [inline Users/Install.rst]: still should be done in a different commit

File: Users/Tutorial.rst (hunk 4 above)
Comment (Adrianzatreanu) [inline Users/Tutorial.rst]: different commit or removed from this one

File: index.rst [hunk 7]
```diff
@@ -115,6 +115,18 @@ Use the ``--format`` option if you want to use a custom format string. Both of
...
+Hint
+~~~~
+
+``--json`` makes the coala output easy to read and understand. It consists of
+a collection of name/value pairs usually represented by objects and an ordered
+list of values stored in arrays. You can read more about this format `here
+<http://www.json.org>`__.
+
+``--format`` returns a linear, one line output. Its string is the most
+customizable and easiest to parse. This option will not show all tested areas
```
Comment (Adrianzatreanu) [inline index.rst]: "its string is the most customizable" this vaguely makes sense tbh

Comment (Adrianzatreanu) [inline index.rst]: all *the* tested areas

**Comment by sils** [comment]
please rebase this, you suffered from a gitmate bug which has been solved yet, that's why all the builds are still yellow. A rebase will solve that.

**Comment by sils** [comment]
ack https://github.com/coala/documentation/pull/370/commits/393e23868a1244bbcccf6a943e3b7e7e1760ad41

**Comment by sils** [comment]
@rultor merge
